src/passes/SafeHeap.cpp

```diff
@@ -100,10 +101,30 @@ struct AccessInstrumenter : public WalkerPass<PostWalker<AccessInstrumenter>> {
   }
 };
 
-struct FindDirectCallees : public WalkerPass<PostWalker<FindDirectCallees>> {
+struct FindCalledFunctions : public WalkerPass<PostWalker<FindCalledFunctions>> {
 public:
-  void visitCall(Call* curr) { callees.insert(curr->target); }
-  std::set<Name> callees;
+  void addFunction(Name name) {
+    functions.insert(name);
+    toVisit.insert(name);
+  }
+
+  void visitCall(Call* curr) {
+    if (functions.count(curr->target) > 0) {
+      return;
+    }
+    addFunction(curr->target);
```

Comment:
Given how simple this operation is, I suggest using `FindAll<Call>`. No need to even define this class etc.

Reply:
Done